pkg/clients/aws.go

```diff
@@ -92,7 +94,8 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 
 	switch s := pc.Spec.Credentials.Source; s { //nolint:exhaustive
 	case runtimev1alpha1.CredentialsSourceInjectedIdentity:
-		return UsePodServiceAccount(ctx, []byte{}, DefaultSection, region)
+		cfg, err := UsePodServiceAccount(ctx, []byte{}, DefaultSection, region)
+		return SetResolver(ctx, mg, cfg), err
```

Comment:
We have an alternative `UseProviderConfigV1` in `master`. Could you add that statement there as well?

Reply:
@muvaf I can't find `UseProviderConfigV1`. What is it?

Reply:
Sorry the wrong name, it's [`GetConfigV1`](https://github.com/crossplane/provider-aws/blob/master/pkg/clients/aws.go#L246). It's an alternative method for controllers that use AWS SDK v1, which are mostly the ones generated by AWS ACK. I can make the addition later if you'd like this to get merged now.

Reply:
I had to implement it as a v1 function cause the versions diverge now on how to handle endpoints but this should do it